### PR TITLE
view extracted contracts refactor

### PR DESCRIPTION
**Pull request by iPR0D1G1**
Fixed: #6567

### CODE REVIEW

File: libs/import/src/lib/components/view-extracted-elements/contract/view-extracted-contracts.component.ts [hunk 1]
```diff
@@ -99,6 +107,47 @@ function getStatic(scope: Scope, value: string) {
   return split(value).map(v => getKeyIfExists(scope, v)).filter(v => !!v);
 }
 
+const fieldsConfig = {
+   /* a */'internationalTitle': (value: string) => {
```
Comment (GrandSchtroumpf) [inline libs/import/src/lib/components/view-extracted-elements/contract/view-extracted-contracts.component.ts:111]: This should be `international.title`

File: libs/import/src/lib/components/view-extracted-elements/contract/view-extracted-contracts.component.ts [hunk 2]
```diff
@@ -99,6 +107,47 @@ function getStatic(scope: Scope, value: string) {
...
+    // const titleId = await getTitleId(value);
+    return new ValueWithWarning(value)
+  },
+   /* b */'contractType': (value: string) => value.toLowerCase(),
```
Comment (GrandSchtroumpf) [inline libs/import/src/lib/components/view-extracted-elements/contract/view-extracted-contracts.component.ts:115]: try to match the real type of the contract, it'll help.

File: libs/import/src/lib/components/view-extracted-elements/contract/view-extracted-contracts.component.ts [hunk 3]
```diff
@@ -99,6 +107,47 @@ function getStatic(scope: Scope, value: string) {
...
+   /* c */'licensorName': (value: string) => value,
+   /* d */'licenseeName': (value: string) => value,
+   /* e */'territoriesList': (value: string) => value,
+   /* f */'mediaList': (value: string) => value,
```
Comment (GrandSchtroumpf) [inline libs/import/src/lib/components/view-extracted-elements/contract/view-extracted-contracts.component.ts:119]: Should be `media` and it should "split" the value to extract an array.

File: libs/import/src/lib/components/view-extracted-elements/contract/view-extracted-contracts.component.ts [hunk 4]
```diff
@@ -99,6 +107,47 @@ function getStatic(scope: Scope, value: string) {
...
+      return new ValueWithWarning(value, errorsMap['no-duration-from']);
+    return date
+  },
+   /* i */'duration.to': (value: string) => {
+    const date = getDate(value);
+    if (isNaN(date.getTime()))
+      return new ValueWithWarning(value, errorsMap['no-duration-to']);
+    return date
+  },
+   /* j ignored*/'originalLanguageLicensed': (value: string) => value,
```
Comment (GrandSchtroumpf) [inline libs/import/src/lib/components/view-extracted-elements/contract/view-extracted-contracts.component.ts:135]: what does "ignored" means in this context ?

File: libs/import/src/lib/components/view-extracted-elements/contract/view-extracted-contracts.component.ts [hunk 5]
```diff
@@ -99,6 +107,47 @@ function getStatic(scope: Scope, value: string) {
...
+   /* k */'dubbed': (value: string) => value,
+   /* l */'subtitle': (value: string) => value,
+   /* m */'closedCaptioning': (value: string) => value,
+   /* n optional*/'contractId': (value: string) => value,
```
Comment (GrandSchtroumpf) [inline libs/import/src/lib/components/view-extracted-elements/contract/view-extracted-contracts.component.ts:139]: what does "optional" means in this context ?

File: libs/import/src/lib/components/view-extracted-elements/contract/view-extracted-contracts.component.ts [hunk 6]
```diff
@@ -160,36 +209,28 @@ export class ViewExtractedContractsComponent implements OnInit {
...
-        dubbed,
-        subtitle,
-        closedCaptioning,
-        _contractId,      // Ok
-        , /* _parentTermId,            ignored lint fix */
-        _titleId,         // Ok
-        _stakeholdersList, // Ok
-      ] = row;
+      const {
+        data: {
```
Comment (GrandSchtroumpf) [inline libs/import/src/lib/components/view-extracted-elements/contract/view-extracted-contracts.component.ts:217]: Don't destructur an object already destructured. It's not readable, and I don't think it's useful at this point.

File: libs/import/src/lib/components/view-extracted-elements/movies/view-extracted-movies.component.ts [hunk 7]
```diff
@@ -32,238 +33,142 @@ import { UserService } from '@blockframes/user/+state';
...
-      subtitle: index++, // BJ
-      caption: index++, // BK
-    }
-  },
-  salesPitch: { // BL
-    multiLine: false,
-    index: index++
-  },
+const fields = [
+  'internationalTitle',
```
Comment (GrandSchtroumpf) [inline libs/import/src/lib/components/view-extracted-elements/movies/view-extracted-movies.component.ts:40]: should be `title.international` I think. Try to match the movie object for all keys.

Comment (GrandSchtroumpf) [reply libs/import/src/lib/components/view-extracted-elements/movies/view-extracted-movies.component.ts:40]: Also I've got th feedling that this array is useless

File: libs/utils/src/lib/spreadsheet/import.ts [hunk 8]
```diff
@@ -20,13 +23,120 @@ export function importSpreadsheet(bytes: Uint8Array, range? :string) : SheetTab[
...
+/**
+* item: The current object to return (contract, movie, org, ...)
+* value: value of the cell (array of string if the cell has several lines)
+* path: The key in the transform object (without the column segment)
+* transform: the callback of the key
+*/
+export function parse(item: any = {}, values: string | string[], path: string, transform: ParseFieldFn, rowIndex: number, warnings:SpreadsheetImportError[], errors:SpreadsheetImportError[]) {
+  // Here we assume the column section has already been removed from the path
+  const segments = path.split('.');
+  // const [segment, ...remainingSegments] = segments;
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/spreadsheet/import.ts:47]: remove comment.

File: libs/utils/src/lib/spreadsheet/import.ts [hunk 9]
```diff
@@ -20,13 +23,120 @@ export function importSpreadsheet(bytes: Uint8Array, range? :string) : SheetTab[
...
+  const segment = segments.shift()
+  const last = !segments?.length;
+  if (segment.endsWith('[]')) {
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/spreadsheet/import.ts:50]: add more comment on the step of the parse function so we can better understand what it does.

File: libs/import/src/lib/components/view-extracted-elements/contract/view-extracted-contracts.component.ts [hunk 10]
```diff
@@ -93,12 +101,55 @@ function getDate(time: string) {
...
+   /* i */'duration.to': (value: string) => getDate(value, errorsMap['no-duration-to']),
+   /* j */'originalLanguageLicensed': (value: string) => value,
+   /* k */'dubbed': (value: string) => getStaticList('languages', value),
+   /* l */'subtitle': (value: string) => getStaticList('languages', value),
+   /* m */'closedCaptioning': (value: string) => getStaticList('languages', value),
+   /* n */'contractId': (value: string) => value,
+   /* o */'parentTermId': (value: string) => value,
+   /* p */'titleId': (value: string) => value,
+   /* q */'stakeholdersList': (value: string) => value ? split(value) : value,
+} as const;
```
Comment (phpgeekfr) [inline libs/import/src/lib/components/view-extracted-elements/contract/view-extracted-contracts.component.ts:145]: this looks like ok

File: libs/import/src/lib/components/view-extracted-elements/movies/view-extracted-movies.component.ts [hunk 11]
```diff
@@ -32,238 +33,88 @@ import { UserService } from '@blockframes/user/+state';
...
+  /* bb */ 'reviews[].quote': (value: string) => value,
+  /* bc */ 'color': (value: string) => value,
+  /* bd */ 'format': (value: string) => value,
+  /* be */ 'formatQuality': (value: string) => value,
+  /* bf */ 'soundFormat': (value: string) => value,
+  /* bg */ 'isOriginalVersionAvailable': (value: string) => value,
+  /* bh */ 'languages[].language': (value: string) => value,
+  /* bi */ 'languages[].dubbed': (value: string) => value,
+  /* bj */ 'languages[].subtitle': (value: string) => value,
+  /* bk */ 'languages[].caption': (value: string) => value,
```
Comment (phpgeekfr) [inline libs/import/src/lib/components/view-extracted-elements/movies/view-extracted-movies.component.ts:106]: you may need to use `getStaticList` like for contract here and maybe for other fields of this list

Comment (iPR0D1G1) [reply libs/import/src/lib/components/view-extracted-elements/movies/view-extracted-movies.component.ts:106]: I just inspectedand seems language, dubbed, subtitle, caption are expected as one off strings and not a character ";" separated string list.

**Pull request merged**